devicetypes/smartthings/ezex-smart-sub-meter.src/ezex-smart-sub-meter.groovy

```diff
@@ -0,0 +1,116 @@
+/**
+ *  Copyright 2019 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+    definition (name: "eZEX Smart Submeter", namespace: "smartthings", author: "SmartThings") {
+        capability "Energy Meter"
+        capability "Power Meter"
+        capability "Refresh"
+        capability "Polling"
+        capability "Sensor"
+
+        fingerprint profileId: "0104", deviceId:"0053", inClusters: "0000, 0003, 0004, 0B04, 0702", outClusters: "0019", manufacturer: "", model: "E240-KR080Z0-HA", deviceJoinName: "eZEX Smart Submeter"
+        
+    }
+
+    // tile definitions
+    tiles(scale: 2) {
+        multiAttributeTile(name:"power", type: "generic", width: 6, height: 4){
+            tileAttribute("device.power", key: "PRIMARY_CONTROL") {
+                attributeState("default", label:'${currentValue} W')
+            }
+            tileAttribute("device.energy", key: "SECONDARY_CONTROL") {
+                attributeState("default", label:'${currentValue} kWh')
+            }
+        }
+        standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", label:'reset kWh', action:"reset"
+        }
+        standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+
+        main (["power", "energy"])
+        details(["power", "energy", "reset", "refresh"])
+    }
+}
+
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        log.info event
+        sendEvent(event)
+    } else {
+        List result = []
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "Desc Map: $descMap"
+          
+        if (descMap.cluster == "0702" && descMap.attrId == "0000") {
```

Comment:
I have the same pair of comments as I had in https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/3975